dcpy/lifecycle/ingest/transform.py

```diff
@@ -109,3 +111,156 @@ def to_parquet(config: Config, local_data_path: Path | None = None):
 
     gdf.to_parquet(PARQUET_PATH, index=False)
     logger.info(f"✅ Converted raw data to parquet file and saved as {PARQUET_PATH}")
+
+
+class Preprocessors:
+    """
+    This class is very much a first pass at something that would support the validate/run_processing_steps functions
+    This should/will be iterated on when implementing actual preprocessing steps for chosen templates
+    """
+
+    @staticmethod
+    def split_column(
+        df: pd.DataFrame,
+        col: str,
+        target_cols: list[str],
+        splitter: Callable[[Any], list[Any]],
+        keep_col=False,
+    ) -> pd.DataFrame:
+        df[target_cols] = df[col].apply(splitter)
+        if not keep_col:
+            df.drop(col, axis=1, inplace=True)
+        return df
+
+    @staticmethod
+    def drop_columns(df: pd.DataFrame, columns: list) -> pd.DataFrame:
+        columns = [df.columns[i] if isinstance(i, int) else i for i in columns]
+        return df.drop(columns, axis=1)
+
+    @staticmethod
+    def strip_columns(df: pd.DataFrame, cols: list[str]) -> pd.DataFrame:
+        if cols == []:
+            df = df.apply(lambda x: x.str.strip() if x.dtype == "object" else x)
+        else:
+            for col in cols:
+                df[col] = df[col].str.strip()
+        return df
+
+    @staticmethod
+    def no_arg_function(df: pd.DataFrame) -> pd.DataFrame:
+        """Dummy/stub for testing. Can be dropped if we implement actual function with no args other than df"""
+        return df
+
+    @staticmethod
+    def append_prev(df: pd.DataFrame, dataset: str, version: str) -> pd.DataFrame:
+        prev_df = recipes.read_df(recipes.Dataset(name=dataset, version=version))
+        df = pd.concat((prev_df, df))
+        return df
+
+    @staticmethod
+    def append_prev_duckdb(file: Path, dataset: str, version: str):
+        duckdb.sql(
+            f"""
+            COPY (
+                SELECT * FROM 's3://edm-recipes/datasets/{dataset}/{version}/{dataset}.parquet'
+                UNION ALL
+                SELECT * FROM '{file}'
+            ) TO '{file}' (FORMAT PARQUET)"""
+        )
+
+
+def validate_function_args(
```

Comment:
Perhaps validation functions like this one should live in `lifecycle`/`ingest`/`metadata`?

Reply:
Potentially. This is a sort of question I'd like to kick down the curb - this function is fairly generic at the moment and theoretically could live outside of `lifecycle` even. Though counterpoint, can't think of an example where we'll need to be validating parsed function arguments anywhere outside of lifecycle.

`lifecycle.ingest.metadata` could also use a rename. I feel like it really is about parsing and validation, or generating configs for these datasets, so something a little more specific to that. Which then still goes in line with your suggestion of moving this there.

So long way of saying sounds good. I do want to rename "metadata" file though. Any thoughts? `config` sort of works, though still a bit generic, would prefer something that's an umbrella for parsing and validation but drawing a blank at the moment

Reply:
Yeah, I agree. It would be nice to have milestone check-ins to evaluate code architecture, and refactor if needed. Versus doing a major refactor in the end. 

Regarding renaming, `config` or `validate` sounds good

Reply:
Done

Reply:
And also broke something in the process of course

Reply:
It's because we use `config` as a var name all over the place.

Given that the other file names are verbs here, I'm going to go with `configure`